### DISCUSSION TITLE
Idea to integrate Fritzbox login script to access smarthome tools like FRITZ!DECT 200

### DISCUSSION THREAD

**Discussion by cool-lox** [arendst/Tasmota#12886] [Ideas]
Hi, 
I found a nice short bash script which can handle login procedure (challenge+PW). With this I could set for example Fritz!Dect 200 power outlet on/off. I'm using it on a R-PI and it works. But may this is a nice idea for implementation in tasmota, opens more options when connection to Fritzbox is done.    (I know power outlets are in Sonoff and easy to switch, but here it is more about integration of Frizbox)

I could share the code but needs to be translated.... :-(
Regards

**Comment by barbudor** [comment]
Just my own perception and personal view.
Tasmota is meant to be flashed on devices and then eventually integrated into an home automation system.
If the Fritz!Dect 200 can't be tasmotized directly (based on ESP chip) then you might used such script to integrate it directly in your home automation.
I don't see the interest of having tasmota in between.

**Comment by cool-lox** [comment]
I know that Fritz!Dect200 can not be tasmotized, this was just an example to switch this device like with own made Fritz!Dect 400 Button based on Wemos D1 with Tasmota and suggested support of Fritzbox login.  May other status can be read out from Fritzbox/device and displayed via ESP8266 and OLED...

**Reply by sfromis** [reply]
How would you envision Tasmota to have a role with that device? Sounds like you want to have Tasmota make a login, and use web scraping? That is pretty far outside of the focus of Tasmota, but of course with open source, if someone comes up with a good implementation....

If I wanted to get "Tasmota compatible" data from such a device, I'd prefer doing it on a more suitable platform, and merely make the MQTT communication similar.

**Comment by cool-lox** [comment]
Again it is just in idea... any yes Tasmota will login in FB and could make as a simple first idea turn on/off such DECT switch. I think when complex login procedure is done once other ideas can be delevoped out of it.

The basis of code can be found on many websites for example here, but Linux code. It works on R-PI in my case.   
https://www.pcwelt.de/ratgeber/Dect-Funksteckdosen-10022547.html
schalter.sh 123456789012 1    (12345678012 is actor id)  and "1" switch on.
But maybe the math behind is too complex to implement in Arduino.

**Reply by sfromis** [reply]
While Tasmota does support bridges without replacing stock firmware on devices (like Zigbee, BLE, IR, RF, KNX), that is to match broad ecosystems around those standards, and tends to provide an easy and low cost path to get these into an MQTT ecosystem. It sounds like "your" FB devices can just as well be supported on the backend server where MQTT and other things already runs, without any extra hardware needed there. Should also be easier to develop than the restrictive environment of a Tasmota driver.

Integrating narrow platforms tends to be a question of someone with a strong wish and skills sitting down to work on it.

**Comment by Jason2866** [comment]
It is a corner case. The goal of Tasmota is to replace the stock firmware in the (cheap) acting device.
Users which bought the expensive FRITZ! solution will probably stay in this world.
If a gateway is wanted/needed it is way easier to do in a Home Automation system.
This can be the same as for Tasmota. Tasmota is designed to be used with (via mqtt).

**Comment by cool-lox** [comment]
ok understood the comments, then we can close it.
